### PR TITLE
move infra kube logic into infra kube manager

### PR DESCRIPTION
**Pull request by arkodg**
Signed-off-by: Arko Dasgupta <arko@tetrate.io>

### CODE REVIEW

File: internal/infrastructure/kubernetes/infra.go [hunk 1]
```diff
@@ -32,11 +38,22 @@ type Resources struct {
...
 func NewInfra(cli client.Client) *Infra {
-	return &Infra{
+	infra := &Infra{
 		mu:        sync.Mutex{},
 		Client:    cli,
 		Resources: newResources(),
 	}
+
+	// Set the namespace used for the managed infra.
+	ns, found := os.LookupEnv("ENVOY_GATEWAY_NAMESPACE")
```
Comment (zirain) [inline internal/infrastructure/kubernetes/infra.go:48]: can use some tool like this https://github.com/istio/pkg/blob/8b4fffa5f37157b1dc5e3aa03b0f88c2a92f55e5/env/var.go#L128

Comment (arkodg) [reply internal/infrastructure/kubernetes/infra.go:48]: its eventually doing the same thing :) https://github.com/istio/pkg/blob/8b4fffa5f37157b1dc5e3aa03b0f88c2a92f55e5/env/var.go#L230
can you clarify how we gain by adopting something similar now ?

Comment (zirain) [reply internal/infrastructure/kubernetes/infra.go:48]: if there's more than one env vars to handle, the code will smell not good.

File: internal/infrastructure/kubernetes/infra.go [hunk 2]
```diff
@@ -13,6 +14,11 @@ import (
 	"github.com/envoyproxy/gateway/internal/ir"
 )
 
+const (
+	// envoyGatewayNamespace is the namespace where envoy-gateway is running.
+	envoyGatewayNamespace = "envoy-gateway-system"
```
Comment (Xunzhuo) [inline internal/infrastructure/kubernetes/infra.go:19]: How about making this const exported? I think envoy-gateway-system will be used not only here in the future. It is like the root namespace of EG.

File: internal/infrastructure/kubernetes/infra.go (hunk 1 above)
Comment (arkodg) [reply internal/infrastructure/kubernetes/infra.go:48]: yah agree, we should def find a way to reuse logic once we have more such env vars

File: internal/infrastructure/kubernetes/infra.go (hunk 2 above)
Comment (arkodg) [reply internal/infrastructure/kubernetes/infra.go:19]: * currently the usage of `envoy-gateway-system` is only in YAML.
* I prefer making the const private now and changing it to visible outside the package once such a case exists

File: internal/infrastructure/kubernetes/infra.go (hunk 1 above)
Comment (danehans) [reply internal/infrastructure/kubernetes/infra.go:48]: > if there's more than one env vars to handle, the code will smell not good.

@zirain feel free to create an issue and add it to the backlog milestone.

**Review by danehans — APPROVED** [review]

**Pull request merged**